docs/docs/development/hardware-integration/shift-registers.md

```diff
@@ -0,0 +1,150 @@
+
+This section can be skipped if you are using the default pins for SPI of a board, e.g. for the Seeed Studio Xiao the pin D8 for the clock signal and D10 for the data signal. However, if you are not making use of the MISO pin and wish to use said pin for alternative purposes, you will need to override the definition.
+
+First, identify the high frequency pins that you are using for SPI by their port and pin number. This example will assume that:
+
+- Your SPI clock pin is port 1, pin 11
+- Your SPI MOSI pin is port 1, pin 12
+
+Next, you'll need to identify the node label of the SPI bus you're overwriting. Look through your board's devicetree files, following the includes, for a node with `compatible = "nordic,nrf-spi";`. This node should have some properties marked as `pinctrl-X`, where `X` is a number. For example, you might find
+
+```dts title="boards/arm/seeeduino_xiao_ble/seeeduino_xiao_ble.dts"
+&spi2 {
+    compatible = "nordic,nrf-spi";
+    pinctrl-0 = <&spi2_default>;
+    pinctrl-1 = <&spi2_sleep>;
+    pinctrl-names = "default", "sleep";
+};
+```
+
+You will need to overwrite the pinctrl nodes to use your desired pins. Look through the devicetree files once again, this time looking for `spi2_default` and `spi2_sleep` (or the equivalently named nodes for your board). You should find something like this:
+
+```dts title="boards/arm/seeeduino_xiao_ble/seeeduino_xiao_ble-pinctrl.dtsi"
+spi2_default: spi2_default {
+    group1 {
+        psels = <NRF_PSEL(SPIM_SCK, 1, 13)>,
+            <NRF_PSEL(SPIM_MOSI, 1, 15)>,
+            <NRF_PSEL(SPIM_MISO, 1, 14)>;
+    };
+};
+
+spi2_sleep: spi2_sleep {
+    group1 {
+        psels = <NRF_PSEL(SPIM_SCK, 1, 13)>,
+            <NRF_PSEL(SPIM_MOSI, 1, 15)>,
+            <NRF_PSEL(SPIM_MISO, 1, 14)>;
+        low-power-enable;
+    };
+};
+```
+
+Overwrite the pin definitions like so:
+
+```dts
+&spi2_default {
+    group1 {
+        psels = <NRF_PSEL(SPIM_SCK, 1, 11)>,
+            <NRF_PSEL(SPIM_MOSI, 1, 12)>;
+    };
+};
+
+&spi2_sleep {
+    group1 {
+        psels = <NRF_PSEL(SPIM_SCK, 1, 11)>,
+            <NRF_PSEL(SPIM_MOSI, 1, 12)>;
+    };
+};
+```
+
+Note that for convenience, ZMK gives the `spi2` node of the Seeed Studio XIAO series the name `xiao_spi`.
+
+:::tip
+If you are making a shield, rather than editing the board's files directly, add a `<your shield folder>/boards/<your board>.overlay` file. This will then be included in your board's definition when you build your shield.
+:::
+
+### Enable SPI
+
+Add the following line to your board's configuration files:
+
+```kconfig title="<your shield folder>/boards/<your board>.conf"
+CONFIG_SPI=y
```

Comment:
I guess this isn't board-specific, right? Can it go in Kconfig.defconfig without issue?

This is what zaphod lite does: https://github.com/petejohanson/zaphod-config/blob/zaphod-lite/boards/shields/zaphod_lite/Kconfig.defconfig#L9

Reply:
Good point, made this change.